### PR TITLE
Add protoc-base image

### PR DESCRIPTION
**Pull request by devinrsmith**
I've tested this locally against a modified deephaven-core and have confirmed that all files under `proto/proto-backplane-grpc/build/generated` are a sha256 match against the original.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -35,6 +26,8 @@ Used as the base image for [deephaven-core's](https://github.com/deephaven/deeph
 
 Produced for platforms amd64 and arm64.
 
+### Web images
```
Comment (niloc132) [inline README.md:29]: should this be `##` instead?

Comment (devinrsmith) [reply README.md:29]: Yes, will fix.

File: README.md [hunk 2]
```diff
@@ -43,7 +36,24 @@ Used as the base image for [deephaven-core's](https://github.com/deephaven/deeph
...
+Used as the base image for building jpy for [deephaven-core](https://github.com/deephaven/deephaven-core).
+There are nice sympathies when building jpy on a system derived from the runtime image.
+
+Produced for platforms amd64 and arm64.
+
+### protoc-base
+
+Based on `node:14-buster-slim`, installs support for JS, python, and java protoc generation.
+
+Produced for platforms amd64.
```
Comment (niloc132) [inline README.md:54]: planning on adding arm? if not, drop s on platforms
> Produced (only) for platfom amd64.

File: proto/01-protoc-base/install.sh [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    unzip
+rm -rf /var/lib/apt/lists/*
+dpkg --list
+
+python3 -m pip install -q --no-cache-dir --upgrade pip
+python3 -m pip install -q --no-cache-dir --upgrade setuptools
+python3 -m pip install -q --no-cache-dir -r "${__dir}/requirements.txt"
+python3 -m pip list
+
+wget -q "https://github.com/protocolbuffers/protobuf/releases/download/v3.15.6/protoc-3.15.6-linux-x86_64.zip"
```
Comment (niloc132) [inline proto/01-protoc-base/install.sh:26]: protoc versions probably should be made consistent between here and package lock

**Review by niloc132 — APPROVED** [review]

File: proto/01-protoc-base/install.sh (hunk 3 above)
Comment (devinrsmith) [reply proto/01-protoc-base/install.sh]: Done. Should also put note or enforce deephaven-core side that it matches up; but that's for follow-up.

After updating protoc, there *were* some changes to the generated code, but only some generated javadoc comments:

```
diff --color -r /tmp/protoc-original/java/io/deephaven/proto/backplane/script/grpc/FigureDescriptorOrBuilder.java /tmp/protoc-new/java/io/deephaven/proto/backplane/script/grpc/FigureDescriptorOrBuilder.java
11c11
<    * <code>string title = 1;</code>
---
>    * <code>optional string title = 1;</code>
16c16
<    * <code>string title = 1;</code>
---
>    * <code>optional string title = 1;</code>
21c21
<    * <code>string title = 1;</code>
---
>    * <code>optional string title = 1;</code>
```

**Pull request merged**